### ISSUE TITLE
contactPoint as uri?

### ISSUE THREAD

**Issue by pvgenuchten** [opengeospatial/ogcapi-records#103]
The current draft indicates to reference a contactPoint as uri. This prevents contactdetails to be repeated multiple times, with risk of typos. However I want to bring in some arguments to introduce explicit contact details in each record (or provide the option to choose, either uri or embedded).
- Not many organisations share their contact details as a vcard in a repository
- vcard is suggested, but if they use schema.org/Person, foaf or even unstructured html or ldap, clients will have a hard time parsing that information
- Organisation-name is an interesting queryable parameter

**Comment by edpars0ns** [comment]
I would support schema.org/Person, indeed I think will should look at schema.org structures where possible - seems a key parameter from a user perspective is accessing the relative value of datasets.

**Comment by pvretano** [comment]
@pvgenuchten how does this prevent contact details from being repeated multiple times with the risks of typos?  If anything wouldn't the opposite be true?  I have a single contact point object that lives somewhere.  I can then reference that contact object in records multiple times using a URL.  Seems to me that would prevent typos from creeping in.  No?
I think your second point if true for any format.  No? You suggest schema.org/Person but if I use VCARD, foaf, HTML, etc.  I will have a hard time parting that information.  That is why VCARD is a suggesting and not a requirement.  B.T.W. VCARD is what DCAT suggested so we went with that.

**Comment by pvgenuchten** [comment]
@pvretano to clarify, I started with listing a benefit of the linkage to a remote resource. But I'm actually doubting if that benefit outweighs the negative aspects of such an approach.

If we suggest an ontology, it means not hardcoding it in to the standard, clients will have a hard time parsing it. So i actually propose to explicitly define the contact properties in the standard. Which ontology to use on top of that is more a conformance class issue, with json-ld-context you're probably able to expose the full record (including contact detail specifics) in the optimal ontology requested by the user.

**Comment by pvretano** [comment]
@pvgenuchten LOL!  My bad ...  for some reason I though you were pointing that out as a bad thing.  Can you sketch up a quick JSON schema for the contact person property for the SWG to consider at the next meeting?  You can either create a PR or simply include your proposal in this issue. Perhaps you can include your initial proposal of it being either a URL or embedded content ... I like that idea.

**Comment by pvgenuchten** [comment]
A proposal for a contact point (inspired by iso19139 & schema.org)

```json
{ "individualname": "",
  "organisationname": "",
  "url": "",
  "email": "",
  "phone": "",
  "postalAddress": {
    "addressCountry": "",
    "addressLocality": "",
    "addressRegion": "",
    "postalCode": "",
    "streetAddress": ""
}
```

or in yaml

```yaml
type: object
required: 
  - organisationname
  - email
  - url
properties:    
  individualname
    type: string
  organisationname
    type: string
  url
    type: string
    format: uri
  email
    type: string
    format: email
  phone
    type: string
    format: phone
  postalAddress:
    type: object
    properties:
      addressCountry
        type: string
      addressLocality
        type: string
      addressRegion
        type: string
      postalCode
        type: string
      streetAddress
        type: string

**Comment by cnreediii** [comment]
If an address is to be encoded as part of "contact", have you considered ISO 19160 Addressing (Parts 1-4) for element names and definitions? Maybe you have already! I looked at the schema.org contactPoint elements and could not find any normative references for where their definitions come from.

**Comment by pvretano** [comment]
08-MAR-2021: @tomkralidis raised the issue that a links is simple to implement but including this makes the core record model more complicated. @cportele in core we should make contact info a URI **OR** an embedded object BUT we should not specify the schema of the object so that we leave the specific details open and not lock them down in the core. @kalxas agreed!
@pvretano will make the changes to the schemas and doc to allow contact into to be a URI or an embedded object.

**Comment by pvretano** [comment]
Added responsibleParty.yaml to schemas.  It allow a contactPoint to the specified by reference as a URI or encoded in-line as a structured object based on ISO 19115 (same was what OWS Common used in the capabilities document ... but in JSON).

**Comment by pvretano** [comment]
03-MAY-2021:  We will wait until the next SWG meeting for feedback and then close since the change has been implemented in the current draft.

**Comment by cholmes** [comment]
Just discovering this now, having an object is definitely a good thing from our experiene in STAC. It'd be great to have more examples of this in the spec, to see how it can work. Also adding what the options are for 'role' - I see  reference to ISO19115:2003, Subclause B.5.5, but I'm not sure how to get that document - it'd be great to just have a set of options to use.

I also couldn't quite tell if multiple 'roles' are allowed? In STAC it's been valuable in our [provider object](https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md#provider-object) to have an organization list several roles. And can you have more than one 'publisher'? In STAC we have an array of provider objects, where you can list multiple organizations. Which is very relevant for how data is often distributed today, where like for landsat on AWS - usgs is the licensor and producer, planet and amazon process it, and amazon is the host. 

A single entity as the sole 'publisher' is just not always the case with data these days.

**Comment by pvgenuchten** [comment]
iso19139 allows only a single role per contact point, but you can add the same person multiple times for each of the roles. iso codelist for role is https://standards.iso.org/iso/19139/resources/gmxCodelists.xml#CI_RoleCode

**Comment by cholmes** [comment]
Awesome, thanks! Would be great to just include that list in this spec. And if those are the only roles then perhaps make the yaml an enum, to validate that people put in a valid role.

**Comment by cholmes** [comment]
Oh, and you answer does seem to imply you can have multiple contact points? Is that right?

**Comment by ByronCinNZ** [comment]
Why are we referencing ISO19115:2003 responsibleParty and not ISO19115-1:2018 CI_Responsibility. The latter 2018 version btw supports ORCid’s

I think we should be more forward looking 

Cheers,
Byron

**Comment by tomkralidis** [comment]
Given the updates in #230 (i.e. a contact object allows for an optional `links` array), should we close this issue? cc @pvretano

**Comment by pvretano** [comment]
01-MAR-2023:  Looks like this issue has been resolved by #230.  Closing.
